rtrefresh/rt_refresh_manager.go

```diff
@@ -202,60 +198,51 @@ func (r *RtRefreshManager) doRefresh(forceRefresh bool) error {
 		merr = multierror.Append(merr, err)
 	}
 
-	for c, lastRefreshedAt := range r.rt.GetTrackedCplsForRefresh() {
-		cpl := uint(c)
-
-		// skip cpls that we've stopped discovering new peers for
-		if r.noResultsForCpl[cpl] >= maxNoResultsAfterRefresh {
-			continue
-		}
-
-		isCplFull := r.rt.IsBucketFull(cpl)
-		peersBeforeRefresh := r.rt.GetPeersForCpl(cpl)
+	refreshCpls := r.rt.GetTrackedCplsForRefresh()
 
-		var err error
+	rfnc := func(c int) (err error) {
```

Comment:
may be worth making forceRefresh / refreshCpls paramters, and making this a method, rather than an anonymous function

Reply:
@willscott Given that this anon func is ONLY used and you'd have to pass it multiple params, I'm not sure we need the new method/abstraction. It's just a simple 4 line func.